app/javascript/widget/store/modules/conversation/actions.js

```diff
@@ -30,6 +30,13 @@ export const actions = {
       commit('setConversationUIFlag', { isCreating: false });
     }
   },
+  setConversationRoutingState: async ({ commit }, status) => {
+    // Handles the routing state during navigation to chat screen
+    // Called before the navigation starts and after navigation completes
+    // Handling this state in app/javascript/widget/router.js
+    // See issue: https://github.com/chatwoot/chatwoot/issues/10736
+    commit('setConversationUIFlag', { isReplacingRoute: status });
```

Comment:
Can we use a more generic name instead of setConversationUIFlag? Since this flag is set on every route change and represents a global routing state that's not specific to conversations, it could be used across multiple screens later.